src/cli/global/upgrade_all.rs

```diff
@@ -33,75 +27,19 @@ pub struct Args {
 }
 
 pub async fn execute(args: Args) -> miette::Result<()> {
-    let packages = list_global_packages().await?;
     let config = Config::with_cli_config(&args.config);
-    let mut channels = config.compute_channels(&args.channel).into_diagnostic()?;
-
-    let mut installed_versions = HashMap::with_capacity(packages.len());
-
-    for package_name in packages.iter() {
-        let prefix_record = find_installed_package(package_name).await?;
-        let last_installed_channel = Channel::from_str(
-            prefix_record.repodata_record.channel.clone(),
-            config.channel_config(),
-        )
-        .into_diagnostic()?;
-
-        channels.push(last_installed_channel);
-
-        let installed_version = prefix_record
-            .repodata_record
-            .package_record
-            .version
-            .into_version();
-        installed_versions.insert(package_name.as_normalized().to_owned(), installed_version);
-    }
-
-    // Remove possible duplicates
-    channels = channels.into_iter().unique().collect::<Vec<_>>();
```

Comment:
I note `channels` is common across all global packages. Is this the intended behaviour?

Reply:
No, I believe the intended behavior is for each global install to use the channels as they were configured at installation time.

Reply:
OK. I'll fix this in a separate PR as this one has been merged.